modular_skyrat/modules/customization/modules/clothing/clothing.dm

```diff
@@ -19,6 +19,130 @@
 	var/greyscale_config_worn_taur_paw
 	var/greyscale_config_worn_taur_hoof
 
+	var/static/has_taur_sprite = list()
+	var/static/has_taur_snake_sprite = list()
+	var/static/has_taur_paw_sprite = list()
+	var/static/has_taur_horse_sprite = list()
+
+	var/static/has_taur_sprite_suit = list()
+	var/static/has_taur_snake_sprite_suit = list()
+	var/static/has_taur_paw_sprite_suit = list()
+	var/static/has_taur_horse_sprite_suit = list()
+
+/obj/item/clothing/Initialize(mapload)
+	. = ..()
+	handle_taur_sprites()
+
+// NOTE FOR BOTH HANDLE_TAUR PROCS: These CAN be converted to handle other things, like digi sprites, or to be ran on init of ANY item. But be aware: For every dependancy you
+// add (digi sprites), you also add a for loop on init, which costs CPU time. The bigger the lists, and the more lists, the more memory we use. We want to use the least amount
+// of lists and loops as possible, and the cache system I set up doesn't completely get rid of the costs.
+
+/**
+ * Called in /obj/item/clothing/Initialize().
+ *
+ * Quits instantly if mutant_variants has all taur styles (STYLE_TAUR), then checks to see if the clothing's icon state has an alt sprite for taurs. If no, it then checks to see if
+ * the icon state EXPLICITELY has no taur sprites. If no, it then checks 3 DMI files for the icon_state, and if it appears in any of them, has_taur_sprite is set to
+ * true for that icon state, then it sets the more specific list to true for that state, and finally the specific mutant_variants bitflag for the clothing if the
+ * clothing doesnt already have it.
+ *
+ * Finally, if all of the conditions are false, has_taur_sprite for that icon state will be set to false.
+ *
+ * DO NOT RELY ON THIS PROC TO DO YOUR WORK FOR YOU. If you are adding a new item, or a sprite, you SHOULD be manually setting your mutant_variants flags, otherwise the for
+ * loop may be ran and cause slight performance loss.
+ */
+/obj/item/clothing/proc/handle_taur_sprites() //niko-if i do not come back in a week and entirely refactor how this works gitban me
+	if (istype(src, /obj/item/clothing/suit))
+		handle_taur_sprites_for_suits()
+		return
+	if (!(istype(src, /obj/item/clothing/under))) //remove this if we ever make a taur sprite for anything else, this is for performance
+		return
+	if (mutant_variants & STYLE_TAUR)
+		return
+	if (has_taur_sprite[icon_state])
```

Comment:
Add an empty newline after each return, please. Or before your if statements, I guess would be more accurate.